client/app/services/dialog-field-refresh.service.js

```diff
@@ -24,6 +24,14 @@
           }
         });
 
+        angular.forEach(allDialogFields, function(dialogField) {
+          angular.forEach(dialogFieldsToRefresh, function(fieldToRefreshName) {
+            if (dialogField.name === fieldToRefreshName) {
+              dialogField.beingRefreshed = true;
+            }
+          });
+        });
```

Comment:
Maybe we don't have to go through all the fields, and use something like `lodash.find(allDialogFields, { name: fieldToRefreshName })` ?